lib/rdf/utils/regex.ex

```diff
@@ -0,0 +1,24 @@
+defmodule RDF.Util.Regex do
+  @moduledoc """
+  Some of the regular expressions in the code base get executed in quote tight
+  loops, sometimes millions of times. The elixir code base does a check on every
+  run of a comppiled regex, to check if the pcre version of the regex matches the
+  version that the local OTP version got compiled against. This check seems to
+  be unexpectedly costly (there is acall though to :erlang.system_info/1). It
+  seems worth it performance wise to curcumvent elixir here, and call trough to 
+  erlang 're' in this case.
+  """
+
+  @doc """
+  Replaces the elixir regex run function. It only works with compiled regexes,
+  and always returns the matched binaries
+  """
+  @spec run(Regex.t(), binary()) :: nil | [binary]
+  def run(%Regex{re_pattern: pattern}, string) do
+    case :re.run(string, pattern, [{:capture, :all, :binary}]) do
+      {:match, matches} -> matches
+      :nomatch -> nil
+      :match -> []
+    end
+  end
+end
```

Comment:
Given the PCRE version check was introduced in Elixir for good reasons, bypassing it generally poses too much of a risk for a library that aims to provide reliable functionality across various environments and Erlang/OTP versions. Bypassing it is only safe in a controlled environment, which doesn't necessarily apply to all users and could lead to issues for them.

To balance offering this optimization while maintaining the library's reliability and compatibility, I propose introducing this optimization as an optional feature enabled through a feature flag configuration option. This allows users, aware of their environment and confident they won't face compatibility issues, to opt-in to this performance improvement, keeping the default library behavior stable and compatible for all users.

Indeed, I found the PR that introduced this version check: https://github.com/elixir-lang/elixir/pull/9040. The linked Elixir core discussion also mentions an alternative implementation without a version check, based on a version cached as a persistent term, which was not implemented at the time because the minimum supported OTP version did not yet support it. It might be worth asking the Elixir core team if, now with OTP 24 as the minimum supported version, it might be reconsidered, especially when faced with benchmark results proving it worthwhile.